src/sine_sweep.cpp

```diff
@@ -35,58 +35,57 @@
 // Original version: Melonee Wise <mwise@willowgarage.com>
 
 #include <math.h>
-#include <control_toolbox/sine_sweep.h>
 
-namespace control_toolbox {
+#include "control_toolbox/sine_sweep.hpp"
 
+namespace control_toolbox
+{
 SineSweep::SineSweep()
+: amplitude_(0.0),
+  duration_(rclcpp::Duration(0, 0)),
+  start_angular_freq_(0.0),
+  end_angular_freq_(0.0),
+  K_(0.0),
+  L_(0.0),
+  cmd_(0.0)
 {
-  K_=0.0;
-  L_=0.0;
-  amplitude_=0.0;
-  duration_ = ros::Duration(0.0);
-  cmd_ = 0.0;
 }
 
-SineSweep::~SineSweep()
-{
-}
+SineSweep::~SineSweep() {}
 
 bool SineSweep::init(double start_freq, double end_freq, double duration, double amplitude)
 {
-  if (start_freq > end_freq)
+  if (start_freq > end_freq) {
     return false;
-  if (duration < 0 || amplitude < 0)
+  }
+  if (duration < 0 || amplitude < 0) {
     return false;
-  
+  }
+
   amplitude_ = amplitude;
-  duration_ = ros::Duration(duration);
-  //calculate the angular fequencies
-  start_angular_freq_ =2*M_PI*start_freq;
-  end_angular_freq_ =2*M_PI*end_freq;
-  
-  //calculate the constants
-  K_ = (start_angular_freq_*duration)/log(end_angular_freq_/start_angular_freq_);
-  L_ = (duration)/log(end_angular_freq_/start_angular_freq_);
-  
-  //zero out the command
+  duration_ = rclcpp::Duration(std::chrono::duration<double>(duration));
```

Comment:
OMFG

I know you are bound by the poor API but could we find a way to pass a double into a duration that is somewhat shorter?

Reply:
`rclcpp::Duration::from_seconds(duration)`

https://github.com/ros2/rclcpp/blob/23719237618ad15f3d074af316cdeabc077f0094/rclcpp/include/rclcpp/duration.hpp#L99-L101

Reply:
thank you :D

Reply:
Thanks!

Reply:
@bmagyar Should be fixed now.